exec/rpc/src/main/java/org/apache/drill/exec/rpc/RemoteConnection.java

```diff
@@ -121,17 +115,72 @@ public void disable() {
     }
   }
 
-  private class BackPressureHandler extends ChannelInboundHandlerAdapter{
+  private class BackPressureHandler extends ChannelInboundHandlerAdapter {
 
     @Override
     public void channelWritabilityChanged(ChannelHandlerContext ctx) throws Exception {
-//      logger.debug("Channel writability changed.", ctx.channel().isWritable());
       writeManager.setWritable(ctx.channel().isWritable());
       ctx.fireChannelWritabilityChanged();
     }
 
   }
 
+  /**
+   * For incoming messages, remove the outcome listener and return it. Can only be done once per coordinationId.
```

Comment:
> can only be done once per coordinationId

Could this be slightly misleading ? `requestIdMap` recycles coordinationIds, so technically this can be called **multiple times** per coordinationId if the Drillbit runs long enough.